src/main/java/robert/storage/Storage.java

```diff
@@ -0,0 +1,136 @@
+     *
+     * @param filePath the file path that stores the data of list of tasks.
+     */
+    public Storage(String filePath) {
+        Path tasksFilePath = Paths.get(System.getProperty("user.dir"), filePath);
+        this.tasksFile = new File(tasksFilePath.toString());
+    }
+
+    /**
+     * Loads the stored tasks from hard disk into Robert.
+     *
+     * @return an ArrayList of Task stored in the hard disk.
+     * @throws RobertException if the stored tasks in hard disk is corrupted or wrongly formatted.
+     */
+    public ArrayList<Task> load() throws RobertException {
+        if (!this.tasksFile.exists()) {
+            this.tasksFile.getParentFile().mkdirs();
+            try {
+                this.tasksFile.createNewFile();
+            } catch (IOException e) {
+                System.out.println(e);
+            }
+        }
+
+        ArrayList<Task> downloadedTasks = new ArrayList<>();
+
+        try (Scanner scanner = new Scanner(this.tasksFile)) {
+            while (scanner.hasNext()) {
+                String[] taskParameters = scanner.nextLine().split(" \\| ");
+
+                switch (taskParameters[0]) {
+                case "T":
+                    ToDo newToDo = new ToDo(taskParameters[2], taskParameters[1].equals("1"));
+                    downloadedTasks.add(newToDo);
+                    break;
+
+                case "E":
+                    LocalDate fromDate = LocalDate.parse(taskParameters[3]);
+                    LocalDate toDate = LocalDate.parse(taskParameters[4]);
+
+                    Event newEvent = new Event(taskParameters[2],
+                            fromDate, toDate, taskParameters[1].equals("1"));
+                    downloadedTasks.add(newEvent);
+                    break;
+
+                case "D":
+                    LocalDate byDate = LocalDate.parse(taskParameters[3]);
+
+                    Deadline newDeadline = new Deadline(taskParameters[2],
+                            byDate, taskParameters[1].equals("1"));
+                    downloadedTasks.add(newDeadline);
+                    break;
+
+                default:
+                    throw new RobertException("An unknown task type was identified "
+                            + "when parsing previously stored tasks.");
+                }
+            }
+        } catch (Exception e) {
+            throw new RobertException("Unable to load data from storage.");
+        }
+
+        return downloadedTasks;
+    }
+
+    /**
+     * Saves the tasks from Robert into the hard disk.
+     *
+     * @throws RobertException if there is an issue saving the tasks into the hard disk.
+     */
+    public void save(TaskList tasks) throws RobertException {
+        try {
+            FileWriter fw = new FileWriter(this.tasksFile.toString(), false);
+            for (Task task : tasks) {
+                String storedLine;
+                String taskDone = task.getStatusIcon().equals("X") ? "1" : "0";
```

Comment:
Since this method has a parameter `TaskList tasks`, you can add a `@param` tag to describe the parameter that is being passed into.

Reply:
Oh I missed this. Thanks a lot!